addons/point_of_sale/static/tests/unit/related_models_test.js

```diff
@@ -0,0 +1,816 @@
+                "Tag 1 should include Product 1, 2 and 3"
+            );
+            assert.ok(readP1.tag_ids.includes(t1, t2), "Product 1 should belong to Tag 1 and 2");
+            const readMany = models["product.product"].readMany([p2.id, p3.id]);
+            assert.deepEqual(readMany, [p2, p3], "Multiple products should be found");
+        });
+
+        QUnit.test("update operation, link", (assert) => {
+            const p1 = models["product.product"].create({});
+            const p2 = models["product.product"].create({});
+            const t1 = models["product.tag"].create({});
+            p1.update({ tag_ids: [["link", t1]] });
+            assert.ok(p1.tag_ids.includes(t1), "Product should be updated with new tag");
+            assert.ok(
+                t1["<-product.product.tag_ids"].includes(p1),
+                "Product should be linked to the new tag"
+            );
+            assert.notOk(
+                t1["<-product.product.tag_ids"].includes(p2),
+                "Product should be unlinked from the old tag"
+            );
+            p2.update({ tag_ids: [["link", t1]] });
+            assert.ok(
+                t1["<-product.product.tag_ids"].includes(p2),
+                "Product should be linked to the tag"
+            );
+        });
+
+        QUnit.test("update operation, unlink", (assert) => {
+            const p1 = models["product.product"].create({});
+            const t1 = models["product.tag"].create({});
+
+            p1.update({ tag_ids: [["link", t1]] });
+            assert.ok(
+                t1["<-product.product.tag_ids"].includes(p1),
+                "Product should be linked to the tag"
+            );
+            assert.ok(p1.tag_ids.includes(t1), "Product should be linked to the tag");
+
+            p1.update({ tag_ids: [["unlink", t1]] });
+            assert.notOk(
+                [t1["<-product.product.tag_ids"]].includes(p1),
+                "Product should be unlinked from the tag"
+            );
+            assert.notOk(p1.tag_ids.length > 0, "Product should be unlinked from the tag");
+            assert.ok(t1, "Product should be linked to the tag");
+        });
+
+        QUnit.test("update operation, Clear", (assert) => {
+            const tag1 = models["product.tag"].create({});
+            const tag2 = models["product.tag"].create({});
+            const product = models["product.product"].create({ tag_ids: [[tag1, tag2]] });
+            models["product.product"].update(product, { tag_ids: [["clear"]] });
+            const updatedproduct = models["product.product"].read(product.id);
+            assert.equal(
+                updatedproduct.tag_ids.length,
+                0,
+                "All tags should be unlinked from the product"
+            );
+            models["product.product"].update(product, { tag_ids: [["link", tag1, tag2]] });
+            assert.ok(product.tag_ids.includes(tag1, tag2), "Product should be linked to the tag");
```

Comment:
I remember commenting about this use of `includes` already. It doesn't support passing multiple items.
```suggestion
            assert.ok([tag1, tag2].every(t => product.tag_ids.includes(t)), "Product should be linked to the tags");
```